chat-ui-manager-app/src/routes/+page.svelte

```diff
@@ -0,0 +1,134 @@
+<script lang="ts">
+	import { onMount } from 'svelte';
+	import { MachineStatus } from '$lib/types';
+	import { env } from '$env/dynamic/public';
+
+	const chatUIUrl = env.PUBLIC_CHAT_UI_URL;
+	let machineStatus: MachineStatus = $state(MachineStatus.UNKNOWN);
+	let chatUIAvailable = $state(false);
+
+	onMount(async () => {
+		await checkMachineStatus();
+	});
+
+	async function checkMachineStatus() {
+		try {
+			const res = await fetch('/api/vm/status');
+			if (!res.ok) throw new Error('Failed to get status');
+			const data = await res.json();
+			machineStatus = data.status; // e.g. MachineStatus.RUNNING or MachineStatus.STOPPED etc.
+		} catch (err) {
+			console.error(err);
+			machineStatus = MachineStatus.UNKNOWN; // revert to 'unknown' if it fails
+		}
+	}
+
+	async function startMachine() {
+		machineStatus = MachineStatus.STARTING;
+		try {
+			const res = await fetch('/api/vm/start', { method: 'POST' });
+			if (!res.ok) throw new Error('Failed to start machine');
+			const data = await res.json();
+			machineStatus = data.status; // e.g. MachineStatus.RUNNING
+		} catch (err) {
+			console.error(err);
+			machineStatus = MachineStatus.STOPPED; // revert if it fails
+		}
+	}
+
+	async function stopMachine() {
+		machineStatus = MachineStatus.STOPPING;
+		chatUIAvailable = false; // Reset availability of the Chat-UI page when stopping the VM
+		try {
+			const res = await fetch('/api/vm/stop', { method: 'POST' });
+			if (!res.ok) throw new Error('Failed to stop machine');
+			const data = await res.json();
+			machineStatus = data.status; // e.g. MachineStatus.STOPPED
+		} catch (err) {
+			console.error(err);
+			machineStatus = MachineStatus.RUNNING; // revert if it fails
+		}
+	}
+
+	// Check if Chat-UI page is available
+	async function checkChatUIAvailability() {
+		try {
+			if (!chatUIUrl) {
+				chatUIAvailable = false;
+				console.error('Chat-UI URL is not defined');
+				return;
+			}
+			const res = await fetch(chatUIUrl, { method: 'GET', mode: 'no-cors' });
```

Comment:
Likewise, you will likely want to cancel this request if after the interval (e.g. 5 seconds) the previous request has not completed (on the odd chance that something that bad happens). Use an `AbortController` :)